File: 0020-valid-parentheses/0020-valid-parentheses.go
func isValid(s string) bool {
    
    stack := []rune{}
    
    closing := map[rune]rune{
        ')':'(',
        '}':'{',
        ']':'[',
    }

    for _, bracket := range s {

        if _, ok := closing[bracket]; !ok {
            stack = append(stack, bracket)
        } else {
            if len(stack) == 0 || stack[len(stack) - 1] != closing[bracket]{
                return false
            }
            stack = stack[:len(stack) - 1]
        }
    }
    return len(stack) == 0
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:

        stack = deque()

        closing = {')':'(', '}':'{',']':'['}

        for bracket in s:
            
            if bracket not in closing:
                stack.append(bracket)

            elif stack and stack[-1] == closing[bracket]:
                stack.pop()

            else:
                return False

        return len(stack) == 0
        